src/rtc_pl031.rs

```diff
@@ -147,52 +161,77 @@ fn get_current_time() -> u32 {
     epoch_time.as_secs() as u32
 }
 
-impl Rtc<NoEvents> {
-    /// Creates a new `AMBA PL031 RTC` instance without any metric
-    /// capabilities.
-    ///
-    /// # Example
-    ///
-    /// You can see an example of how to use this function in the
-    /// [`Example` section from `Rtc`](struct.Rtc.html#example).
-    pub fn new() -> Rtc<NoEvents> {
-        Self::with_events(NoEvents)
-    }
-}
-
 impl Default for Rtc<NoEvents> {
     fn default() -> Self {
         Self::new()
     }
 }
 
-impl<EV: RtcEvents> Rtc<EV> {
-    /// Creates a new `AMBA PL031 RTC` instance and invokes the `rtc_events`
-    /// implementation of `RtcEvents` during operation.
-    ///
-    /// # Arguments
-    /// * `rtc_events` - The `RtcEvents` implementation used to track the occurrence
-    ///                  of failure or missed events in the RTC operation.
-    pub fn with_events(rtc_events: EV) -> Self {
-        Rtc {
+// This is the state from which a fresh Rtc can be created.
+impl Default for RtcState {
+    fn default() -> Self {
+        RtcState {
             // The load register is initialized to 0.
             lr: 0,
-
             offset: 0,
-
             // The match register is initialised to zero (not currently used).
             // TODO: Implement the match register functionality.
             mr: 0,
-
             // The interrupt mask is initialised as not set.
             imsc: 0,
-
             // The raw interrupt is initialised as not asserted.
             ris: 0,
+        }
+    }
+}
 
+impl Rtc<NoEvents> {
+    /// Creates a new `AMBA PL031 RTC` instance without any metric
+    /// capabilities.
+    pub fn new() -> Self {
+        Self::from_state(&RtcState::default(), NoEvents)
+    }
+}
+
+impl<EV: RtcEvents> Rtc<EV> {
+    /// Creates a new `AMBA PL031 RTC` instance from a given `state` and invokes the `rtc_events`
+    /// implementation of `RtcEvents` during operation.
+    ///
+    /// # Arguments
+    /// * `state` - A reference to the state from which the `Rtc` is constructed.
+    /// * `rtc_events` - The `RtcEvents` implementation used to track the occurrence
+    ///                  of failure or missed events in the RTC operation.
+    pub fn from_state(state: &RtcState, events: EV) -> Self {
+        Rtc {
+            lr: state.lr,
+            offset: state.offset,
+            mr: state.mr,
+            imsc: state.imsc,
+            ris: state.ris,
             // A struct implementing RtcEvents for tracking the occurrence of
             // significant events.
-            events: rtc_events,
+            events,
+        }
+    }
+
+    /// Creates a new `AMBA PL031 RTC` instance and invokes the `rtc_events`
+    /// implementation of `RtcEvents` during operation.
```

Comment:
The part about "invokes the `rtc_events`" sounds like a implementation detail, and it shouldn't be part of the interface documentation. How about: "Creates a new `AMBA PL031 RTC` instance that is able to track events during operation using the passed `rtc_events` object." We should also add a paragraph about the state being the default one or something like that.